### ISSUE TITLE
Update for Visual Studio 2022

### ISSUE THREAD

**Issue by ddobrev** [jzoss/Git-Source-Control-Provider#128]
It isn't even listed as available.

**Comment by asine** [comment]
need too

**Comment by NightOwl888** [comment]
I also need the VS2022 update. I would contribute the PR if I had the bandwidth. Maybe one of you could submit it?

**Comment by maciey** [comment]
Same here - I do also miss that in vs2022

**Comment by youxiaotian** [comment]
need too

**Comment by jzoss** [comment]
Hey all..sorry It's not listed.. I'll take a look at see if I can't get it in there

**Comment by jzoss** [comment]
Also sorry...for some reason I stopped getting notifications on this

**Comment by jzoss** [comment]
> I also need the VS2022 update. I would contribute the PR if I had the bandwidth. Maybe one of you could submit it?

If you have time to do a pr.. I'd totally take it

**Comment by jzoss** [comment]
HI all.. I am running into a bunch of issues with the upgrade.. this is a big one.. I'll keep going.. but again.. it anyone wants to help :)

**Comment by ceztko** [comment]
I'm sorry to be here just to say there's still desperate need for your extension in VS2022. The bundled integration in VS looks to be of no use to me: it's unable to switch on different repositories based on the focused file and it doesn't have a quick diff view.

**Comment by maciey** [comment]
@jzoss any luck with this yet ?

**Comment by NightOwl888** [comment]
I managed to update [Git Tools](https://github.com/yysun/git-tools/issues/40#issuecomment-1311816638) to work with VS2022.

There is a [helpful VS2022 extension upgrade guide here](https://learn.microsoft.com/en-us/visualstudio/extensibility/migration/update-visual-studio-extension?view=vs-2022).

`<UseCodebase>true</UseCodebase>` also seems to be required in order for the menus to show up in VS2022. Be mindful that none of the dependencies that you are referencing require 32 bit, as VS2022 no longer supports it. It is also best to use NuGet packages instead of local references, if they exist.

**Comment by maciey** [comment]
Thanks for updating this! Long waited !
However I cannot install it as it says it requires VS Common Project system.
I cant finda a way how to install that - can you plz advise ?

**Comment by jzoss** [comment]
Awesome.. I am going to test this out. Thanks.. I

**Comment by jzoss** [comment]
@NightOwl888 Did you update this fork or the original fork? Is there a pull request I can see?

**Comment by NightOwl888** [comment]
Nope, I updated Git Tools (a different plugin). See https://github.com/yysun/git-tools/pull/41.

But the same type of issues will apply here, so hopefully I have provided you with a roadmap to upgrade this tool.

**Comment by NightOwl888** [comment]
> Thanks for updating this! Long waited ! However I cannot install it as it says it requires VS Common Project system. I cant finda a way how to install that - can you plz advise ?

Hmm... I [added that into the metadata](https://github.com/yysun/git-tools/pull/41/commits/63533d91d7f37fd5da94c11604737b3cb1949247#diff-4917b9b690afe78054a7943806234c6e649e1e7fb198a7cdf5286c90c967910eR20) because the VS 2019 release had it. I have no idea if it is actually a prerequisite of the tool. You might want to try compiling without it to see if that works. Otherwise, you might need to research how to use Visual Studio installer to install the required prerequisite. Either way, it would be helpful if you report your findings to https://github.com/yysun/git-tools/ if there is something about the tool that needs to be changed.

**Comment by jzoss** [comment]
@NightOwl888 I've tried that page before and got stuck. But I can give it another go..
